Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -111,10 +111,24 @@ - (BOOL)canBecomeFirstResponder {
     return YES;
 }
 
+- (BOOL)pasteBoardHasText
+{
+    if ([UIPasteboard generalPasteboard].numberOfItems < 1) {
+        return NO;
+    }
+    NSIndexSet *itemSet = [NSIndexSet indexSetWithIndex:0];
+    NSSet<NSString *> *utiTypes =
+        [NSSet setWithArray:[[UIPasteboard generalPasteboard] pasteboardTypesForItemSet:itemSet][0]];
+    return ([utiTypes containsObject:(NSString *)kUTTypeText] || [utiTypes containsObject:(NSString *)kUTTypePlainText]
+        ||
+        [utiTypes containsObject:(NSString *)kUTTypeUTF8PlainText] ||
+        [utiTypes containsObject:(NSString *)kUTTypeUTF16PlainText]);
+}
+
 - (BOOL)pasteBoardHasPossibleAttachment {
     // We don't want to load/convert images more than once so we
     // only do a cursory validation pass at this time.
-    return [SignalAttachment pasteboardHasPossibleAttachment];
+    return ([SignalAttachment pasteboardHasPossibleAttachment] && ![self pasteBoardHasText]);
```

Comment:
I wanted to be precise about considering only the first item if the pasteboard contains multiple items, since that's all we'll use.

In the long run, it'd be nice to support pasting multiple items at once.